gdal/swig/include/ogr.i

```diff
@@ -1652,6 +1651,15 @@ public:
       }
   }
 %clear GByte*;
+#elif defined(SWIGCSHARP)
+%apply (void *buffer_ptr) {char **};
+  OGRErr GetFieldAsBinary( int id, int *nLen, char **pBuf) {
+    GByte* pabyBlob = OGR_F_GetFieldAsBinary(self, id, nLen);
+    *pBuf = (char*)malloc(*nLen);
```

Comment:
I believe this will leak.  You'd likely need something similar to the ``%typemap(freearg) (int *nLen, char **pBuf )`` of swig/include/java/typemaps_java.i to free it

And I guess you could probably avoid the manual public byte[] GetFieldAsBinary(int id) override by having an output typemap that would return a byte[], but I'm not familiar with C# swig typemaps to provide more guidance

Reply:
Ok - back to the drawing board. I will look at that

Reply:
Apologies, I've missed following up this conversation since having success with the code @runette provided. I don't have the technical skills to fix this myself, but it seems close to being complete, apart from the 'suspected' memory leak issue. Is that something a more experienced OSGeo contributor could quickly fix up to get this included - possibly @szekerest or @bjornharrtell?